spec/services/nudge_mail_spec.rb

```diff
@@ -0,0 +1,106 @@
+    let!(:user_2) { create(:user, :registered, confirmed_at: 4.weeks.ago) }
+    let!(:user_3) { create(:user, :registered, confirmed_at: 4.weeks.ago) }
+    let!(:user_4) { create(:user, :registered, confirmed_at: 2.months.ago) }
+    let!(:user_5) { create(:user, :registered, confirmed_at: 4.weeks.ago, module_time_to_completion: { "alpha": 0 }) }
+
+    before do
+      allow(NotifyMailer).to receive(:start_training)
+    end
+
+    it 'The notify mailer is called with the correct users' do
+      nudge_mail.call
+      expect(NotifyMailer).to have_received(:start_training).with(user_1).once
+      expect(NotifyMailer).to have_received(:start_training).with(user_2).once
+      expect(NotifyMailer).to have_received(:start_training).with(user_3).once
+      expect(NotifyMailer).not_to have_received(:start_training).with(user_4)
+      expect(NotifyMailer).not_to have_received(:start_training).with(user_5)
+    end
+  end
+
+  context 'when users are a month old and have completed registration and started training but not completed training' do
+    include_context 'with progress'
+    let!(:user_1) { create(:user, :registered, confirmed_at: 4.weeks.ago, module_time_to_completion: { "alpha": 0 }) }
+    let!(:user_2) { create(:user, :registered, confirmed_at: 4.weeks.ago, module_time_to_completion: { "alpha": 0 }) }
+    let!(:user_3) { create(:user, :registered, confirmed_at: 4.weeks.ago, module_time_to_completion: { "alpha": 0 }) }
+    let!(:user_4) { create(:user, :registered, confirmed_at: 4.weeks.ago, module_time_to_completion: { "alpha": 1 }) }
+    let!(:user_5) { create(:user, :registered, confirmed_at: 8.weeks.ago, module_time_to_completion: { "alpha": 2 }) }
+
+    before do
+      users = [user_1, user_2, user_3]
+      users.each do |user|
+        course_progress = Rails.application.cms? ? instance_double(ContentfulCourseProgress) : instance_double(CourseProgress)
+        allow(CourseProgress).to receive(:new).with(user: user).and_return(course_progress)
+        allow(ContentfulCourseProgress).to receive(:new).with(user: user).and_return(course_progress)
+        allow(course_progress).to receive(:current_modules).and_return([Training::Module.by_name('alpha')])
+      end
+
+      Ahoy::Visit.new(
+        id: 1,
+        visitor_token: '123',
+        user_id: user_1.id,
+        started_at: 4.weeks.ago,
+      ).save!
+
+      Ahoy::Visit.new(
+        id: 2,
+        visitor_token: '234',
+        user_id: user_1.id,
+        started_at: 2.weeks.ago,
+      ).save!
+
+      Ahoy::Visit.new(
+        id: 3,
+        visitor_token: '345',
+        user_id: user_2.id,
+        started_at: 4.weeks.ago,
+      ).save!
+
+      Ahoy::Visit.new(
+        id: 4,
+        visitor_token: '456',
+        user_id: user_3.id,
+        started_at: 4.weeks.ago,
+      ).save!
+
+      allow(NotifyMailer).to receive(:continue_training)
+    end
+
+    it 'The notify mailer is called with the correct users' do
```

Comment:
Not essential but it can be useful to write as if the audience is not a developer. 

`it 'emails...`

Reply:
[62482be](https://github.com/DFE-Digital/early-years-foundation-recovery/pull/726/commits/62482befe8bddaab78d25d93bcc6384777f18953)